youtube_dl/extractor/youmaker.py

```diff
@@ -0,0 +1,196 @@
+                    (?:v|video|embed)/
+                    (?P<id>[0-9a-zA-Z-]+)
+                    """
+
+    _TESTS = [
+        {
+            "url": "http://www.youmaker.com/v/71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+            "info_dict": {
+                "id": "71b5d2c5-31b6-43b8-8475-1dcb5e10dfb0",
+                "ext": "mp4",
+                "title": "Как сшить шапочку из трикотажа. Плоский шов двойной иглой.",
+                "description": r"re:(?s)^Привет друзья!\n\nВ этом видео я .* представлена www\.iksonmusic\.com$",
+                "thumbnail": r"re:^https?://.*\.(?:jpg|png)$",
+                "duration": 358,
+                "upload_date": "20190614",
+                "uploader": "user_318f21e00e1f8a6b414f20a654d0f4fc7d2053bc",
+                "timestamp": 1560554895,
+                "channel": "Sewing Ideas",
+                "channel_id": "40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "channel_url": r"re:https?://www.youmaker.com/channel/40ca79f7-8b21-477f-adba-7d0f81e5b5fd",
+                "tags": [
+                    "как сшить детскую шапочку из трикотажа",
+                    "как шить двойной иглой трикотаж",
+                ],
+                "categories": ["Life", "How-to & DIY"],
+            },
+            "params": {
+                "skip_download": True,
+            },
+        },
+    ]
+
+    def __init__(self, downloader=None):
+        """Constructor. Receives an optional downloader."""
+        super(YouMakerIE, self).__init__(downloader=downloader)
+        self.__protocol = "https"
+        self.__video_id = None
+        self.__cache = {}
+
+    def _set_id_and_protocol(self, url):
+        self.__video_id = self._match_id(url)
+        if url.startswith("http://"):
+            self.__protocol = "http"
+
+    def _fix_url(self, url):
+        if url.startswith("//"):
+            return "%s:%s" % (self.__protocol, url)
+        return url
+
+    @property
+    def _base_url(self):
+        return self._fix_url("//www.youmaker.com")
+
+    @property
+    def _asset_url(self):
+        # as this url might change in the future
+        # it needs to be extracted from some js magic...
+        return self._fix_url("//vs.youmaker.com/assets")
+
+    def _api(self, path, cache=False, **kwargs):
+        """
+        call the YouMaker JSON API and return the data
+
+        path:       API endpoint
+        **kwargs:   parameters passed to _download_json()
+        """
+        key = hash((path, compat_urllib_parse_urlencode(kwargs.get("query", {}))))
+        if cache and key in self.__cache:
+            return self.__cache[key]
+
+        url = "%s/v1/api/%s" % (self._base_url, path)
+        info = self._download_json(url, self.__video_id, **kwargs)
+        status = info.get("status", "something went wrong")
+        data = info.get("data")
+        if status != "ok" or data is None:
+            raise ExtractorError(status, expected=True)
```

Comment:
Above, it's still possible that the JSON download works but doesn't result in a `dict`. So this would be better (setting `None` on error to help with the second point below):
```
        status = try_get(info, lambda x: x['status'])
```
Then, if the API makes a breaking change without us noticing, is that `expected` or not? As the site is unlikely to revert the change, it becomes our bug and so not `expected`. I suggest `expected` should correspond to the API returning an actual status that is not OK, and nothing else, like so:
```
             raise ExtractorError(status or 'something went wrong', expected=status not in ('ok', None))
```
But you are obviously familiar with the API and I'm not ...

Reply:
status and data are minimal requirements. And don't worry, if expected errors happen repeatedly people will bother you anyway :)